api/src/handlers/helpers/retrieveAndEncryptFiles.ts

```diff
@@ -0,0 +1,30 @@
+import { InputFields } from "./getAllInputsFromForm";
+import { EncryptionService, FileService } from "../../middlewares/services";
+import { MailAttachments } from "../../types/MailAttachments";
+
+export async function retrieveAndEncryptFiles(
```

Comment:
I think your file and encryption services can be in one place. encryption service is encrypting _files_.

I don't think this function is necessary though. FWIW if I still need this function, I would refactor this so this function only handles one at a time, and use a promise aggregate like `Promise.allSettled`. I think `Promise.all` is most appropriate though. You want all of them to succeed, and throw if not.

```
const uploadFields = [];

const filePromises = uploadFields.map(retrieveAndEncrypt);
try {
  const encryptedFiles = await Promise.all(filePromises);
  
} catch (e) {


}
```